### PR TITLE
Rubocop style linting

### PR DESCRIPTION
**Pull request by alexander-griffen**
## Relevant issue(s)
N/A
## What does this do?
The following RuboCops:
- [AndOr](https://rubystyle.guide#no-and-or-or)
- [NumericLiteralPrefix](https://rubystyle.guide#numeric-literal-prefixes)
- [TrailingCommaInArrayLiteral](https://rubystyle.guide#no-trailing-array-commas)
- IfInsideElse
- [PreferredHashMethods](https://rubystyle.guide#hash-key)
- ZeroLengthPredicate
- [Lambda](https://rubystyle.guide#lambda-multi-line)
- [RedundantBegin](https://rubystyle.guide#begin-implicit)

### CODE REVIEW

File: app/controllers/concerns/fragment_cachable.rb [hunk 1]
```diff
@@ -37,6 +37,6 @@ def foi_fragment_cache_write(key_path, content)
     File.atomic_write(key_path) do |f|
       f.write(content)
     end
-    FileUtils.chmod 0644, key_path
+    FileUtils.chmod 0o644, key_path
```
Comment (garethrees) [inline app/controllers/concerns/fragment_cachable.rb:40]: Hmm I don't quite understand this check but -1 without exploring this more.

`FileUtils.chmod 0644` is much clearer than `FileUtils.chmod 0o644` to me.

Some of the changes where we're parsing dates don't look right.

File: app/models/holiday_import.rb [hunk 2]
```diff
@@ -95,14 +95,14 @@ def populate_from_ical_event(cal_event)
...
+    
       holiday_info = Holidays.between(start_date, end_date, @country_code.to_sym, :observed)
       holiday_info.each do |holiday_info_hash|
         holidays << Holiday.new(description: holiday_info_hash[:name],
                                 day: holiday_info_hash[:date])
       end
     rescue Holidays::InvalidRegion
       []
-    end
+
```
Comment (garethrees) [inline app/models/holiday_import.rb:106]: Where it removes these redundant begin/end blocks, the contents should also be shifted left to get back to two-spacing.

Simple example:

```rb
# BEFORE
def foo
  begin
    bar
  rescue SomeError
    # do something
  end
end

# AFTER AUTOCORRECT
def foo
    bar
  rescue SomeError
    # do something
end

# WHAT WE WANT
def foo
  bar
rescue SomeError
  # do something
end
```

Comment (alexander-griffen) [reply app/models/holiday_import.rb]: This has been changed and squashed into that commit

File: lib/mail_handler/backends/mail_backend.rb [hunk 3]
```diff
@@ -327,17 +325,15 @@ def _get_attachment_leaves_recursive(part, within_rfc822_attachment, parent_mail
                                                                parent_mail)
             end
           end
-        else
+        elsif part.rfc822_attachment
           # Add all the parts of a decoded attached message
-          if part.rfc822_attachment
-            leaves_found += _get_attachment_leaves_recursive(part.rfc822_attachment,
+          leaves_found += _get_attachment_leaves_recursive(part.rfc822_attachment,
```
[the remark this answers is not in the document]
Comment (garethrees) [reply lib/mail_handler/backends/mail_backend.rb:330]: The `else` just needs to be indented -2 to the left

File: app/controllers/concerns/fragment_cachable.rb (hunk 1 above)
Comment (alexander-griffen) [reply app/controllers/concerns/fragment_cachable.rb]: The commit has been removed, and the cop disabled (for now).

**Review by gbp — APPROVED** [review]
One fixup for indentation for the `IfInsideElse` changes

**Pull request merged**